File: src/main/java/com/github/daniel/shuy/kotlin/spring/boot/example/dto/PetDto.java
package com.github.daniel.shuy.kotlin.spring.boot.example.dto;

import com.github.daniel.shuy.kotlin.spring.boot.example.model.Status;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.Collection;
import lombok.Builder;

@Builder
public record PetDto(
    Long id,
    @NotBlank String name,
    @NotNull Status status,
    Collection<String> tags) {
}

File: src/main/kotlin/com/github/daniel/shuy/kotlin/spring/boot/example/dto/PetDto.kt
package com.github.daniel.shuy.kotlin.spring.boot.example.dto

import com.github.daniel.shuy.kotlin.spring.boot.example.model.Status
import jakarta.validation.constraints.NotBlank
import kotlinx.serialization.Serializable

@Serializable
data class PetDto(
    val id: PetId? = null,
    @field:NotBlank val name: String,
    val status: Status,
    val tags: Collection<String>? = null,
)

typealias PetId = Long
